src/overlays/actors/ovl_En_Fd_Fire/z_en_fd_fire.c

```diff
@@ -21,23 +28,216 @@ const ActorInit En_Fd_Fire_InitVars = {
+void EnFdFire_Disappear(EnFdFire* this, GlobalContext* globalCtx) {
+    Math_SmoothScaleMaxMinF(&this->actor.speedXZ, 0.0f, 0.6f, 9.0f, 0.0f);
+    func_8002D868(&this->actor);
+    Math_SmoothScaleMaxMinF(&this->scale, 0.0f, 0.3f, 0.1f, 0.0f);
+    this->actor.shape.unk_10 = 20.0f;
+    this->actor.shape.unk_10 *= (this->scale / 3.0f);
+    if (!(this->scale > 0.01f)) {
+        Actor_Kill(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fd_Fire/func_80A0E70C.s")
+void EnFdFire_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnFdFire* this = THIS;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fd_Fire/func_80A0E848.s")
+    if (this->actionFunc != EnFdFire_Disappear) {
+        if ((this->actor.parent->update == NULL) || EnFdFire_CheckCollider(this, globalCtx)) {
+            this->actionFunc = EnFdFire_Disappear;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fd_Fire/func_80A0E884.s")
+    func_8002D7EC(&this->actor);
+    this->actionFunc(this, globalCtx);
+    func_8002E4B4(globalCtx, &this->actor, 12.0f, 10.0f, 0.0f, 5);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fd_Fire/func_80A0EA34.s")
+    if (this->actionFunc != EnFdFire_Disappear) {
+        Collider_CylinderUpdate(&this->actor, &this->collider);
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fd_Fire/EnFdFire_Update.s")
+void EnFdFire_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    Color_RGBA8 primColors[] = {
+        { 255, 255, 0, 255 },
+        { 255, 255, 255, 255 },
+    };
+    Color_RGBA8 envColors[] = {
+        { 255, 10, 0, 255 },
+        { 0, 10, 255, 255 },
+    };
+    s32 pad;
+    EnFdFire* this = THIS;
+    Vec3f scale = { 0.0f, 0.0f, 0.0f };
+    Vec3f sp90 = { 0.0f, 0.0f, 0.0f };
+    s16 sp8E;
+    f32 sp88;
+    f32 sp84;
+    f32 sp80;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fd_Fire/EnFdFire_Draw.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_fd_fire.c", 0x23C);
+    Matrix_Translate(this->actor.posRot.pos.x, this->actor.posRot.pos.y, this->actor.posRot.pos.z, MTXMODE_NEW);
+    sp8E = Math_Vec3f_Yaw(&scale, &this->actor.velocity) - func_8005A9F4(ACTIVE_CAM);
+    sp84 = fabsf(Math_Coss(sp8E));
+    sp88 = Math_Sins(sp8E);
+    sp80 = Math_Vec3f_DistXZ(&scale, &this->actor.velocity) / 1.5f;
+    if (1) {}
+    if (1) {}
+    if (1) {}
+    Matrix_RotateY((s16)(func_8005A9F4(ACTIVE_CAM) + 0x8000) * (M_PI / 32768.0f), MTXMODE_APPLY);
+    Matrix_RotateZ(((sp88 * -10.0f) * sp80) * (M_PI / 180.0f), MTXMODE_APPLY);
+    scale.x = scale.y = scale.z = this->scale * 0.001f;
+    Matrix_Scale(scale.x, scale.y, scale.z, MTXMODE_APPLY);
+    sp84 = sp80 * (-0.14999999f * sp84) + 1.0f;
```

Comment:
```suggestion
    sp84 = sp80 * (-15 * 0.01 * sp84) + 1.0f;
```
Could also be something like -15.0/100. These off-by-one floats are usually the result of double conversions from calculated literals like this.